### ISSUE TITLE
Updating affected components for a case is hard/confusing

### ISSUE THREAD

**Issue by mjtko** [alces-software/alces-flight-center#839]
When updating affected components for a case, it's not immediately clear what components can be associated with the current case given its current issue type.  An multi-step procedure is required:

1. Modify issue type
2. Update affected components

However, in order to discover that this is necessary, preceding failed attempts almost certainly occur.

Couple of thoughts:

1. If components can't be associated with the current issue type, they shouldn't be shown
2. Modify issue type/components at the same time with appropriate visual cues about what's valid and what's not
3. Don't care which components can be associated with a case after it's already open -- the main reason for having limits with components and case types is to guide users into creating the case in the first place.

**Comment by jamesremuscat** [comment]
There are no circumstances I can think of where a component cannot be associated with a case because of its issue type. The only restriction is that certain issue types require at least a certain type of service and/or component (e.g. the dependency is being described the wrong way around in the above).

For example, if you have a "Suspected hardware issue" case, it must have _at least one_ "Hardware Management" service and _at least one_ component associated. As long as those conditions are satisfied, other services and components may be associated/disassociated freely. If the issue type is changed to, say, "Other (related to service)" then the only restrictions are that _at least one_ service (of any type) must be associated.

Hence, either the set of components affected by 1. is empty, or else you mean the significantly stronger "we should make it that only components/services explicitly required by the current issue type may be associated with a case / are shown" - I don't think that's what you mean, but correct me if I'm wrong.

We don't currently explicitly display an issue type's requirements (other than through error messages). Perhaps doing so, and perhaps doing so on the "change associations" page, would make it more obvious that the association change you're about to make would violate them? Bringing the functionality to change the two together could also work (I'd need to think about the best way to display them).

Does it make sense to allow a "suspected hardware issue" case to have no hardware associated with it? If not then we should continue to prevent that scenario. If we don't care then 3. is a valid option and probably the least effort to implement.

**Comment by mjtko** [comment]
TL;DR - seems option 3 is the least effort and most flexible, and we should trust admins to be able to Do The Right Thing :tm:

In more detail:

You're right that it doesn't make sense for a suspected hardware issue to have no hardware associated with it however, at the moment, the process of changing associated components and issue type (when a user has been unable to find the right place to log a case for example) needs quite a bit of understanding of what issue types require what kind of components.

For example, if a case has been logged as a software issue, and it's actually a hardware issue, there is a three-step process which involves:

1. Add hardware service (and possibly hardware components)
2. Change issue type
3. Remove HPC environment service

It seems to me that the easiest approach is to leave it up to the admin what components are associated with a case post-creation given that they should understand what makes sense.

**Comment by jamesremuscat** [comment]
Option 3 implemented in #946.
